src/main/java/org/example/DeliveryApp.java

```diff
@@ -0,0 +1,102 @@
+package org.example;
+import java.io.File;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+/**
+ * Small logistics application that simulates package delivery.
+ */
+public class DeliveryApp {
+    static List<Package> packages = new ArrayList<>();
+
+    public static void main(String[] args) throws InterruptedException {
+        DeliveryApp app = new DeliveryApp();
+        Scanner scanner = null;
+        try {
+            scanner = new Scanner(
+                    new File("input.csv"));
+            while (scanner.hasNextLine()) {
+                List<String> line = parseLine(scanner.nextLine());
+                app.packages.add(app.createPackage(line));
+            }
+        } catch (IOException e) {
+            System.out.println("File not found");
+            System.exit(0);
+        } finally {
+            if (scanner != null) {
+                scanner.close();
+            }
+        }
+        Collections.sort(packages, new LocationComparator());
```

Comment:
how do you test this sorting?

Reply:
also line 33 does a grouping but the method is called listToSet, no verb, nothing about stuff being grouped